pkg/sbom/cyclonedx/core/cyclonedx.go

```diff
@@ -289,17 +295,23 @@ func (c *CycloneDX) Licenses(licenses []string) *cdx.Licenses {
 	return lo.ToPtr(cdx.Licenses(choices))
 }
 
-func (c *CycloneDX) Properties(props map[string]string) []cdx.Property {
-	properties := lo.MapToSlice(props, func(k, v string) cdx.Property {
-		return cdx.Property{
-			Name:  Namespace + k,
-			Value: v,
+func (c *CycloneDX) Properties(properties []Property) []cdx.Property {
+	cdxProps := make([]cdx.Property, 0)
+	for _, property := range properties {
+		namespace := Namespace
+		if len(property.Namespace) > 0 {
+			namespace = property.Namespace
 		}
-	})
+		cdxProps = append(cdxProps,
+			cdx.Property{
+				Name:  namespace + property.Name,
+				Value: property.Value,
+			})
+	}
 	sort.Slice(properties, func(i, j int) bool {
 		return properties[i].Name < properties[j].Name
 	})
```

Comment:
You renamed `properties` to `cdxProps`. This sorting also needs to be updated.

```suggestion
	sort.Slice(cdxProps, func(i, j int) bool {
		return cdxProps[i].Name < cdxProps[j].Name
	})
```

Reply:
thanks , good catch